Doc/library/types.rst

```diff
@@ -409,7 +409,9 @@ Additional Utility Classes and Functions
                return "{}({})".format(type(self).__name__, ", ".join(items))
 
            def __eq__(self, other):
-               return self.__dict__ == other.__dict__
+               if isinstance(other, SimpleNamespace):
```

Comment:
You could do this:
```
>>> SimpleNamespace.__eq__(3, SimpleNamespace())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor '__eq__' requires a 'types.SimpleNamespace' object but received a 'int'
```
So I suggest leaving in the `isinstance` check for `self` to more closely match what the C code is doing.

Reply:
Thanks for your patience!